src/test/java/org/openrewrite/java/migrate/lang/var/UseVarForGenericMethodInvocationsTest.java

```diff
@@ -259,5 +260,40 @@ void m() {
               )
             );
         }
+
+        @Issue("https://github.com/openrewrite/rewrite-migrate-java/issues/862")
+        @Test
+        void nestedGenericsSimple() {
+            //language=java
+            rewriteRun(
+              version(
+                java(
+                  """
+                    import java.util.ArrayList;
+                    import java.util.Collections;
+                    import java.util.List;
+
+                    class A {
+                      void dostuff() {
+                          List<String> strs = Collections.synchronizedList(new ArrayList<>());
```

Comment:
Are there additional cases we might need to cover here you think?

Reply:
I could think of factory methods that might need special handling, like `List<Inetger> ints = List.of()` --> `var ints = List.<Integer>of()` but if I recall correctly, we decided to not migrate static generic methods at all, due to this.

Reply:
I wonder why we haven't skipped `Collections.synchronizedList(..)` in the first place; it's a generic static method. Would you mind verifying if you find any loophole there?

Reply:
all good here, `Collections.synchronizedList(..)` is migrated because it has parameters, `List.of()` is not as it has no parameters.